Source/monstdat.h

```diff
@@ -61,17 +61,299 @@ enum class MonsterClass : uint8_t {
+	 * @return True if any of the monster's non-acid resistances are greater than 0.
+	 */
+	bool hasResistancesOrImmunities() const
+	{
+		return this->Magic > 0 || this->Fire > 0 || this->Lightning > 0;
+	}
+	/**
+	 * @brief Returns if monster has any resistances to Magic, Fire, or Lightning. Must have at least one greater than 0 but less than 100.
+	 * @return True if any of the monster's non-acid resistances are greater than 0 but less than 100.
+	 */
+	bool hasResistances() const
+	{
+		return (this->Magic > 0 && this->Magic < 100) || (this->Fire > 0 && this->Fire < 100) || (this->Lightning > 0 && this->Lightning < 100);
+	}
+
+	/**
+	 * @brief Returns if monster has any immunities to Magic, Fire, or Lightning.
+	 * @return True if monster has non-acid immunities.
+	 */
+	bool hasImmunities() const
+	{
+		return this->Magic >= 100 || this->Fire >= 100 || this->Lightning >= 100;
+	}
+
+	/**
+	 * @brief Returns if monster has resistance (not immunity) to Magic.
+	 * @return True if monster has resistance greater than 0 but less than 100 to Magic.
+	 */
+	bool isMagicResistant() const
+	{
+		return this->Magic > 0 && this->Magic < 100;
+	}
+
+	/**
+	 * @brief Returns if monster has immunity to Magic.
+	 * @return True if monster has resistance greateer than or equal to 100 to Magic.
+	 */
+	bool isMagicImmune() const
+	{
+		return this->Magic >= 100;
+	}
+
+	/**
+	 * @brief Returns if monster has resistance (not immunity) to Fire.
+	 * @return True if monster has resistance greater than 0 but less than 100 to Fire.
+	 */
+	bool isFireResistant() const
+	{
+		return this->Fire > 0 && this->Fire < 100;
+	}
+
+	/**
+	 * @brief Returns if monster has immunity to Fire.
+	 * @return True if monster has resistance greateer than or equal to 100 to Fire.
+	 */
+	bool isFireImmune() const
+	{
+		return this->Fire >= 100;
+	}
+
+	/**
+	 * @brief Returns if monster has resistance (not immunity) to Lightning.
+	 * @return True if monster has resistance greater than 0 but less than 100 to Lightning.
+	 */
+	bool isLightningResistant() const
+	{
+		return this->Lightning > 0 && this->Lightning < 100;
+	}
+
+	/**
+	 * @brief Returns if monster has immunity to Lightning.
+	 * @return True if monster has resistance greateer than or equal to 100 to Lightning.
+	 */
+	bool isLightningImmune() const
+	{
+		return this->Lightning >= 100;
+	}
+
+	/**
+	 * @brief Returns if monster has resistance (not immunity) to Acid.
+	 * @return True if monster has resistance greater than 0 but less than 100 to Acid.
+	 */
+	bool isAcidResistant() const
+	{
+		return this->Acid > 0 && this->Acid < 100;
+	}
+
+	/**
+	 * @brief Returns if monster has immunity to Acid.
+	 * @return True if monster has resistance greateer than or equal to 100 to Acid.
+	 */
+	bool isAcidImmune() const
+	{
+		return this->Acid >= 100;
+	}
+
+	/**
+	 * @brief Sets all monster resistances to 0.
+	 */
+	void setToZero()

```

Comment:
name doesn't explain anything (should contain the word resistances or something)

Reply:
Considering you have to be calling it from something directly related to resistances, I disagree that this is entirely unclear, however I will modify accordingly. Thank you!

Reply:
nvm, I didn't see it was used for resistances only :D a bit weird design